pkg/datapath/l2responder/l2responder.go

```diff
@@ -114,36 +116,46 @@ func (p *l2ResponderReconciler) cycle(
 	fullReconciliation <-chan time.Time,
 ) {
 	arMap := p.params.L2ResponderMap
+	ndMap := p.params.L2V6ResponderMap
 	log := p.params.Logger
 
 	lr := cachingLinkResolver{nl: p.params.NetLink}
 
 	process := func(e *tables.L2AnnounceEntry, deleted bool) error {
-		// Ignore IPv6 addresses, L2 is IPv4 only
-		if e.IP.Is6() {
-			return nil
-		}
 
 		idx, err := lr.LinkIndex(e.NetworkInterface)
 		if err != nil {
 			return fmt.Errorf("link index: %w", err)
 		}
 
 		if deleted {
-			err = arMap.Delete(e.IP, uint32(idx))
+			if e.IP.Is6() {
+				err = arMap.Delete(e.IP, uint32(idx))
+			} else {
+				err = ndMap.Delete(e.IP, uint32(idx))
+			}
```

Comment:
Perhaps im misreading this, but you define 

```go
arMap := p.params.L2ResponderMap
ndMap := p.params.L2V6ResponderMap
``` 

shouldn´t this therefore be the other way around?
```suggestion
			if e.IP.Is6() {
				err = ndMap.Delete(e.IP, uint32(idx))
			} else {
				err = arMap.Delete(e.IP, uint32(idx))
			}
``

Reply:
Indeed, thx. I plan to (finally) continue working on this.